code/Controllers/CMSMain.php

```diff
@@ -1984,12 +1965,9 @@ public function delete($data, $form)
     /**
      * Delete this page from both live and stage
      *
-     * @param array $data
-     * @param Form $form
-     * @return HTTPResponse
      * @throws HTTPResponse_Exception
      */
-    public function archive($data, $form)
+    public function archive(array $data, Form $form): HTTPResponse
```

Comment:
This method sometimes returns `Security::permissionFailure()` which sometimes returns an array

Reply:
As noted above, it does not return an array